sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeColumnCommand.scala

```diff
@@ -50,7 +50,8 @@ case class AnalyzeColumnCommand(
           AnalyzeTableCommand.calculateTotalSize(sessionState, catalogRel.catalogTable))
 
       case logicalRel: LogicalRelation if logicalRel.catalogTable.isDefined =>
-        updateStats(logicalRel.catalogTable.get, logicalRel.relation.sizeInBytes)
+        updateStats(logicalRel.catalogTable.get,
+          AnalyzeTableCommand.calculateTotalSize(sessionState, logicalRel.catalogTable.get))
```

Comment:
This is needed since the logical relation `sizeInBytes` is no longer the latest size. Previously ListingFileCatalog would compute the size, but TableFileCatalog just reads whatever size was present in the table metadata last.